src/components/SeldonImage/SeldonImage.tsx

```diff
@@ -0,0 +1,117 @@
+import { ComponentProps, forwardRef, useRef, useState, useEffect } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+
+type AspectRatio = '16/9' | '1/1' | 'none';
+
+export interface SeldonImageProps extends ComponentProps<'div'> {
+  /**
+   * The aspect ratio of the image container.
+   */
+  aspectRatio?: AspectRatio;
+  /**
+   * The resize behavior of the image (see CSS object-fit).
+   */
+  objectFit?: 'contain' | 'cover' | 'fill' | 'none' | 'scale-down';
+  /**
+   * Whether the image has a blur background covering the background of the image container
+   */
+  hasBlurBackground?: boolean;
+  /**
+   * The image to display.
+   */
+  src: string;
+  /**
+   * The alt text of the image.
+   */
+  alt: string;
+  /**
+   * The class name of the child img element.
+   */
+  imageClassName?: string;
+  /**
+   * The style of the child img element.
+   */
+  imageStyle?: React.CSSProperties;
+}
+
+/**
+ * ## Overview
+ *
+ * Component for displaying an image with optional blur background, aspect ratio, and object fit.
+ *
+ * [Figma Link](https://www.figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?m=auto&node-id=4501-64590)
+ *
+ * [Storybook Link](https://phillips-seldon.netlify.app/?path=/docs/components-seldonimage--overview)
+ */
+const SeldonImage = forwardRef<HTMLDivElement, SeldonImageProps>(
+  (
+    {
+      className,
+      aspectRatio = 'none',
+      objectFit = 'none',
+      hasBlurBackground = false,
+      imageClassName,
+      imageStyle,
+      src,
+      alt,
+      ...props
+    },
+    ref,
+  ) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'SeldonImage');
+    const imgRef = useRef<HTMLImageElement>(null);
+
+    const [loadingState, setLoadingState] = useState<'loading' | 'loaded' | 'error'>('loading');
+
+    useEffect(() => {
+      if (imgRef.current?.complete) {
+        setLoadingState('loaded');
+      }
+    }, []);
```

Comment:
Why did we need this in addition to the `onLoad` below? I suspect this won't actually get fired since the `imgRef.current` is never set on initial render.

Reply:
we may not, I think I threw this in there to fix this weird flash I was seeing in browser but it may have just been a browser cache thing, and it was making it seem like `onLoad` was firing inconsistently

it actually is defined though, since the first `useEffect` fires after first paint